SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+				$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "WSA"
+				$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue ($wsaTmpObject | Select-Object -Skip 0)
+				$passwordPolicyNodes.Add($nodeConfigValue)
+				return $passwordPolicyNodes
+			} else {
+				Write-LogMessage -Type Error -Message "Failed to obtain access token from WSA server '$server',  please verify your credentials and try again"
+			}
+		} else {
+				Write-LogMessage -Type Error -Message "Testing a connection to server $server failed, please check your details and try again"
+		}
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+}
+
+Function GetEnvironmentPasswordPolicyDetail {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [psobject]$ppmVariables,
+        [Parameter (Mandatory = $true)] [AllowEmptyCollection()][System.Collections.Generic.List[System.Object]]$ppmEnvironmentalDetails
+	)
+	if ($ppmVariables.isEnvDetailSet -eq $false) {
+		Write-LogMessage -type INFO -Message "Retrieving Environment Details....(this might take a while depends on the number of esxi hosts)" -Colour Yellow
+		
+		##Retrieve ESXi hosts password policy details
+		Write-LogMessage -type INFO -Message "Retrieving ESXi Hosts password policies.."
+		$vcenter = Get-vCenterServerDetail -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName
+		$clusterID = Get-VCFWorkloadDomain -name $ppmVariables.sddcDomainName
+		$ppmVariables.esxiCluster = $clusterID
+		foreach ($cid in $ppmVariables.esxiCluster.clusters.id) {
+			$cluster = Get-VCFCluster -id $cid
+			$results = GetEsxiPasswordPolicy -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -cluster $cluster.name
+			foreach ($result in $results) {
+				$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+				$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+				$ppmEnvironmentalDetails.Add($result)
+			}
+		}
+		Write-LogMessage -type INFO -Message "Retrieving ESXi Hosts password policies..complete"
+		
+		##Retrieve VC password policy details 
+		Write-LogMessage -type INFO -Message "Retrieving VC server password policies.."
+		$result = GetVCServerPasswordPolicy -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName
+		$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+		$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+		$ppmEnvironmentalDetails.Add($result)
+		Write-LogMessage -type INFO -Message "Retrieving VC server password policies..complete"
```

Comment:
"vCenter Server instance...."

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e